common/src/main/java/org/opensearch/ml/common/transport/mcpserver/requests/register/McpTool.java

```diff
@@ -0,0 +1,136 @@
+/*
+ *
+ *  * Copyright OpenSearch Contributors
+ *  * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package org.opensearch.ml.common.transport.mcpserver.requests.register;
+
+import static org.opensearch.core.xcontent.XContentParserUtils.ensureExpectedToken;
+
+import java.io.IOException;
+import java.util.Map;
+
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+
+import lombok.Data;
+import lombok.extern.log4j.Log4j2;
+
+/**
+ *  This class represents a tool that can be registered with OpenSearch. It contains information about the tool's name,
+ * description, parameters, and schema.
+ */
+@Log4j2
+@Data
+public class McpTool implements ToXContentObject, Writeable {
+    private static final String NAME_FIELD = "name";
+    private static final String DESCRIPTION_FIELD = "description";
+    private static final String PARAMS_FIELD = "params";
+    private static final String SCHEMA_FIELD = "schema";
+    private final String name;
+    private final String description;
+    private Map<String, Object> params;
+    private Map<String, Object> schema;
```

Comment:
Suggest keep consistent with current tool definition, for example
```
    {
      "type": "IndexMappingTool",
      "attributes": {
        "input_schema": {
          "type": "object",
          "properties": {
            "index": {
              "type": "array",
              "description": "OpenSearch index name list, separated by comma. for example: index1, index2",
              "items": {
                "type": "string"
              }
            }
          },
          "required": ["index"],
          "additionalProperties": false
        },
        "strict": true
      }
    }
```

Reply:
Done